compaction/compaction_manager.cc

```diff
@@ -2294,8 +2294,10 @@ compaction_manager::maybe_split_sstable(sstables::shared_sstable sst, compaction
     if (!split_compaction_task_executor::sstable_needs_split(sst, opt)) {
         co_return std::vector<sstables::shared_sstable>{sst};
     }
-    if (!can_proceed(&t)) {
-        co_return std::vector<sstables::shared_sstable>{sst};
+    // This procedure can bypass check if view has compaction disabled, since this operation is offline,
+    // meaning the sstable is not loaded into any sstable set yet. And it's required for correctness.
+    if (is_disabled()) {
+        throw std::runtime_error(format("Unable to split SSTable {} because compaction manager is disabled", sst->get_filename()));
```

Comment:
Alternatively, we allow repairing sstables to be compacted between themselves. @asias do you remember why it takes so long for a new repaired sstable to be marked as repaired? it runs in background right?

Reply:
A sstable produced by repair will be marked as repaired only if the everything goes well and there is no error on all the peers. They are marked together with the sstables being selected for repair.